### PR TITLE
Initial setup for cucumber framework

### PR DESCRIPTION
**Pull request by karthikmsqa**
# Description

This PR includes the implementation of cucumber framework for both frontend and backend tests. 

- Completes Jira https://justice.gov.bc.ca/jira/browse/FLA-103

## Type of change

- [x] New test(non-breaking change which adds tests)
- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

on localhost

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ tests/logs
 *.iml
 *.log
 .idea
-
+tests/test-output
+*.png
```
Comment (akroon3r) [inline .gitignore:9]: missing newline

File: tests/README.md [hunk 2]
```diff
@@ -31,3 +31,25 @@ Installs all the required mvn dependencies.
...
 
 Runs the tests on the specified browser.
+
+## Running the Cucumber Tests with tags
+Add values to the keys in config.properties file
+Update tags value in 'RunCucumberTest' class
+
+In the 'tests' directory, you can run:
+
+### `mvn test -Dcucumber.options="--tags '@tagname'"`
```
Comment (akroon3r) [inline tests/README.md:41]: I think this should be ```bash BLAH ``` for a bash command

Comment (akroon3r) [reply tests/README.md:41]: triple back tick COMMAND triple back tick...sorry didn't anticipate formatting there

File: tests/README.md [hunk 3]
```diff
@@ -31,3 +31,25 @@ Installs all the required mvn dependencies.
...
+
+This runs the tests that are tagged with the specific tag name. 
+Without the 'tags' option all tests gets executed.
+
+Alternatively, both frontend and backend tests can be run from 
+
+### `RunCucumberTest.java` class
```
Comment (akroon3r) [inline tests/README.md:48]: I think these shouldn't be headings...? Alex might want to chime in. We were talking about readme formatting earlier in the day.

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/backend/generateurlpayload/GenerateUrlPayload.java [hunk 4]
```diff
@@ -1,27 +1,23 @@
...
+        Headers headers = new Headers(STRING, STRING, STRING);
         SubmissionAccess submissionAccess = new SubmissionAccess(STRING,"GET", headers );
         DocumentProperties documentProperties = new DocumentProperties(STRING, STRING, submissionAccess);
         Navigation navigation = new Navigation(success,error,cancel);
+
         Payload payload = new Payload(documentProperties, navigation);
 
         ObjectMapper objMap = new ObjectMapper();
-        return objMap.writerWithDefaultPrettyPrinter().writeValueAsString(payload);
+            return objMap.writerWithDefaultPrettyPrinter().writeValueAsString(payload);
```
Comment (akroon3r) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/backend/generateurlpayload/GenerateUrlPayload.java:21]: formatting looks weird in the diff here for some reason.

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import io.cucumber.junit.Cucumber;
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+         features = {"src/test/java/features"},
+         glue ={"stepDefinitions"},
+        // tags = "@backend",
```
Comment (akroon3r) [inline tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java:11]: if commented out, do we need this? or we can mark for a TODO?

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+         monochrome = true,
+         plugin = {
+                 "pretty",
+                 "html:target/cucumber-reports/cucumber-pretty",
+                 "json:target/cucumber-reports/CucumberTestReport.json",
+                 "rerun:target/cucumber-reports/rerun.txt",
+                 "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"
+                }
+                )
```
Comment (akroon3r) [inline tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java:20]: formatting looks weird here again, not sure if its the diff or not.

File: tests/src/test/java/features/backendfeatures/generateUrl.feature [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Feature: Secure url can be generated as per the api contract
+
+  @backend
+  Scenario: Verify if the secure url is generated
+    Given user calls efilingAPI with POST http request
+    When status code and content type are verified
+    Then the response returns efiling url and expiry date
```
Comment (akroon3r) [inline tests/src/test/java/features/backendfeatures/generateUrl.feature:7]: missing newline

File: tests/src/test/java/stepDefinitions/frontendstepdefinitions/LandingPageTest.java [hunk 8]
```diff
@@ -43,4 +39,10 @@ public void verifyLandingPageTitleTest() {
...
         log.info("Correct page title is verified.");
     }
-}
+
+    @And("browser is closed")
+    public void browser_is_closed() {
+        driver.quit();
+        log.info("Browser closed");
+    }
+}
```
Comment (akroon3r) [inline tests/src/test/java/stepDefinitions/frontendstepdefinitions/LandingPageTest.java:48]: missing newline

File: tests/src/test/resources/extent-config.xml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                });
+                ]]>
+        </scripts>
+        <!-- custom styles -->
+        <styles>
+            <![CDATA[
+            ]]>
+        </styles>
+    </configuration>
+</extentreports>
```
Comment (akroon3r) [inline tests/src/test/resources/extent-config.xml:43]: missing newline, if its auto generated its all good

**Comment by akroon3r** [comment]
I'll fix the small things tomorrow morning since Karthik is on time off. :)

File: .gitignore (hunk 1 above)
Comment (karthikmsqa) [reply .gitignore]: Fixed. Thanks!

File: tests/README.md (hunk 2 above)
Comment (karthikmsqa) [reply tests/README.md:41]: Fixed. Thanks!

File: tests/README.md (hunk 3 above)
Comment (karthikmsqa) [reply tests/README.md]: Fixed. Thanks!

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java (hunk 5 above)
Comment (karthikmsqa) [reply tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java]: Fixed. Thanks!

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java (hunk 6 above)
Comment (karthikmsqa) [reply tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/cucumber/Options/RunCucumberTest.java]: Fixed. Thanks!

File: tests/src/test/java/stepDefinitions/frontendstepdefinitions/LandingPageTest.java (hunk 8 above)
Comment (karthikmsqa) [reply tests/src/test/java/stepDefinitions/frontendstepdefinitions/LandingPageTest.java]: Fixed. Thanks!

File: tests/src/test/resources/extent-config.xml (hunk 9 above)
Comment (karthikmsqa) [reply tests/src/test/resources/extent-config.xml]: Fixed. Thanks!

File: tests/src/test/java/features/backendfeatures/generateUrl.feature (hunk 7 above)
Comment (karthikmsqa) [reply tests/src/test/java/features/backendfeatures/generateUrl.feature]: Fixed. Thanks!

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/backend/generateurlpayload/GenerateUrlPayload.java (hunk 4 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/backend/generateurlpayload/GenerateUrlPayload.java]: Fixed. Thanks!

**Pull request merged**